### PR TITLE
[orchestrator] [process] isolating orchestrator settings

### PR DESCRIPTION
**Pull request by nammn**
### What does this PR do?
Isolating orchestrator settings out of the `process_config`

### Motivation

Isolate orchestrator related topics from process.

### Describe your test plan

Unit Tests + manual test.
Manual test:
- Run the Process-agent with the old settings (deprecated, but backward compatible)
    `process_config.orchestrator_additional_endpoints` and `process_config.orchestrator_dd_url` 

- Run the Process-agent with the new settings 
    `orchestrator_explorer.orchestrator_additional_endpoints` and `orchestrator_explorer.orchestrator_dd_url`.

verify in logs/backend whether data is send.

### CODE REVIEW

File: pkg/process/config/yaml_config_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
+package config
+
+import (
+	coreConfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/stretchr/testify/suite"
+	"net/url"
+	"testing"
+)
+
+type YamlConfigTestSuite struct {
```
Comment (nammn) [inline pkg/process/config/yaml_config_test.go:10]: The tests under "config" seems to test more setting "envs" and general the `config` file. Whereas here the test are more under the `yaml_config` file. I could put them together, but I like the easy usage of a suite 🤔

File: pkg/process/config/yaml_config_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	u, _ := url.Parse("https://test.com")
+	actualEndpoints := []api.Endpoint{{APIKey: "test", Endpoint: u}}
+
+	suite.config.Set("api_key", "wassupkey")
+	suite.config.Set("process_config.orchestrator_additional_endpoints", `{"https://process1.com": ["key1"], "https://process2.com": ["key2", "key3"]}`)
+	suite.config.Set("orchestrator_explorer.orchestrator_additional_endpoints", `{"https://orchestrator1.com": ["key1"], "https://orchestrator2.com": ["key2", "key3"]}`)
+	err := extractOrchestratorAdditionalEndpoints(&url.URL{}, &actualEndpoints)
+	suite.NoError(err)
+	for _, actual := range actualEndpoints {
+		suite.Equal(expected[actual.APIKey], actual.Endpoint.Hostname(), actual)
```
Comment (hkaj) [inline pkg/process/config/yaml_config_test.go:111]: what's the third `actual` argument for?

File: pkg/config/config.go [hunk 3]
```diff
@@ -635,12 +634,20 @@ func InitConfig(config Config) {
...
 	// enabling/disabling the environment variables & command scrubbing from the container specs
 	// this option will potentially impact the CPU usage of the agent
 	config.BindEnvAndSetDefault("orchestrator_explorer.container_scrubbing.enabled", true)
 
+	// Orchestrator Explorer - process agent
+	config.BindEnv("orchestrator_explorer.orchestrator_dd_url", "") //nolint:errcheck
+	// DEPRECATED in favour of `orchestrator_explorer.orchestrator_dd_url` setting. If both are set `orchestrator_explorer.orchestrator_dd_url` will take precedence.
+	config.BindEnv("process_config.orchestrator_dd_url", "") //nolint:errcheck
+	config.SetKnown("process_config.orchestrator_additional_endpoints.*")
+	// DEPRECATED in favour of `orchestrator_explorer.orchestrator_additional_endpoints` setting. If both are set `orchestrator_explorer.orchestrator_additional_endpoints` will take precedence.
```
[the remark this answers is not in the document]
Comment (nammn) [reply pkg/config/config.go:648]: Oh, yes. Thanks for the catch!

File: pkg/process/config/yaml_config_test.go (hunk 2 above)
Comment (nammn) [reply pkg/process/config/yaml_config_test.go:111]: Oh damn some copy pasta. Thanks for the catch!

**Review by hkaj — APPROVED** [review]

**Pull request merged**